src/emulator/storage/files.ts

```diff
@@ -149,14 +240,58 @@ export class StorageLayer {
    * @throws {ForbiddenError} if request is unauthorized
    */
   public async getObject(request: GetObjectRequest): Promise<GetObjectResponse> {
+    let authorized = true;
     const metadata = this.getMetadata(request.bucketId, request.decodedObjectId);
 
     // If a valid download token is present, skip Firebase Rules auth. Mainly used by the js sdk.
     const hasValidDownloadToken = (metadata?.downloadTokens || []).includes(
       request.downloadToken ?? ""
     );
-    let authorized = hasValidDownloadToken;
-    if (!authorized) {
+
+    let checkAuth = false;
+
+    if (!hasValidDownloadToken) {
+      if (request.urlSignature) {
+        const prevSignature = request.urlSignature;
+
+        if (!request.urlUsableSeconds || !request.urlTtlSeconds) {
+          throw new BadRequestError(`Invalid ${!request.urlUsableSeconds ? "Date" : "TTL"}`);
```

Comment:
It'd be great if this error contained the name of the missing parameter. For instance, if `urlUsableSeconds` is missing, the error should read:

```
Missing required parameter X-Firebase-Date: YYYYMMDD'T'HHMMSS'Z'
```

Similar for urlTtlSeconds.